### DISCUSSION TITLE
其他裝置無法連到在localhost 運行的Vue 專案

### DISCUSSION THREAD

**Discussion by TehoChang** [Lidemy/forum#224] [General]
老師好：
碰到一個問題卡了非常久，連問題的方向都還沒確立，故來求救~ 
我在本機運行一個vue專案，碰到以下情形
- 本機透過localhost:8000/#/ 可以正常連線，但是透過http://192.168.1.107:8000 連線不到，
- 其他專案沒有這問題

確定有做到以下行為
1 關閉mac防火牆 2 確定裝置在同一個wifi

後來終於找到一點端倪，發現兩個專案運行時有以下的不同
![messageImage_1634739443126](https://user-images.githubusercontent.com/61130008/138112435-0bc54ca9-9f84-42bb-899a-e19d01509344.jpg)
![messageImage_1634739462944](https://user-images.githubusercontent.com/61130008/138112444-0797260d-f6d2-412e-b90e-dae13d7568a3.jpg)

也就是說，運行後沒有顯示出ip的那個專案， 就無法用http://192.168.1.107 連線
但是找了一段時間，還沒找到為什麼會這樣？ 要在哪邊設定？ 會是在Vue.config檔案嗎
懇請有經驗的大大分享～ 感謝。

**Comment by aszx87410** [comment]
我 google `vue server listen ip`
找到這個：https://segmentfault.com/q/1010000014797068

**Reply by TehoChang** [reply]
謝謝老師！根據這篇文章的資訊，找到vue.config.js中的配置項 devServer，出問題的專案的devServer內容如下

```tsx
devServer: {
    port: port,
    host: 'localhost',
    open: true,
    overlay: {
      warnings: true,
      errors: true
    }
  },
```
  其中`host: 'localhost'` 這一項在運作正常的專案中是沒有的。將該行刪除後，問題解決
